src/container/HashContainer/HashSet.ts

```diff
@@ -1,55 +1,93 @@
-import { initContainer } from '@/container/ContainerBase';
-import HashContainerBase from '@/container/HashContainer/Base';
+import { initContainer, IteratorType } from '@/container/ContainerBase';
+import { HashContainer, HashContainerIterator } from '@/container/HashContainer/Base';
+import $checkWithinAccessParams from '@/utils/checkParams.macro';
+import { throwIteratorAccessError } from '@/utils/throwError';
 
-class HashSet<K> extends HashContainerBase<K, undefined> {
+class HashSetIterator<K, V> extends HashContainerIterator<K, V> {
+  get pointer() {
+    if (this._node === this._header) {
+      throwIteratorAccessError();
+    }
+    return this._node._key;
+  }
+  copy() {
+    return new HashSetIterator(this._node, this._header, this.iteratorType);
+  }
+  // @ts-ignore
+  equals(iter: HashSetIterator<K, V>): boolean;
```

Comment:
Why do you use ts-ignore?

Reply:
TS tell me it must have a function body.